src/compiler/program.ts

```diff
@@ -2061,6 +2193,31 @@ namespace ts {
             return allFilesBelongToPath;
         }
 
+        function parseProjectReferenceConfigFile(ref: ProjectReference): { commandLine: ParsedCommandLine, sourceFile: SourceFile } | undefined {
+            // The actual filename (i.e. add "/tsconfig.json" if necessary)
+            const refPath = resolveProjectReferencePath(host, ref);
+            // An absolute path pointing to the containing directory of the config file
+            const basePath = getNormalizedAbsolutePath(getDirectoryPath(refPath), host.getCurrentDirectory());
+            const sourceFile = host.getSourceFile(refPath, ScriptTarget.JSON) as JsonSourceFile;

```

Comment:
Why not use: ```getParsedCommandLineOfConfigFile``` instead ?

Reply:
Feedback from Mohamed was that we want to pass around `SourceFile`s instead of filenames so that in-memory editing of referenced projects would work